.github/workflows/publish-prerelease.yml

```diff
@@ -0,0 +1,72 @@
+name: Publish prerelease
+
+on:
+  workflow_call:
+    secrets:
+      PR_COMMENT_TOKEN:
+        required: true
+
+jobs:
+  publish-prerelease:
+    name: Publish prerelease
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # This is needed to get merge base to calculate bundle size diff
+
+      - name: Setup environment
+        uses: metamask/github-tools/.github/actions/setup-environment@main
+
+      - name: Get merge base commit hash
+        id: get-merge-base
+        env:
+          BASE_REF: ${{ github.event.pull_request.base.ref }}
+        run: |
+          merge_base="$(git merge-base "origin/${BASE_REF}" HEAD)"
+          echo "Merge base is '${merge_base}'"
+          echo "MERGE_BASE=${merge_base}" >> "$GITHUB_OUTPUT"
+
+      - name: Get CircleCI job details
+        id: get-circleci-job-details
+        env:
+          REPOSITORY: ${{ github.repository }}
+          BRANCH: ${{ github.head_ref }}
+          HEAD_COMMIT_HASH: ${{ github.event.pull_request.head.sha }}
+        run: |
+          pipeline_id=$(curl --silent "https://circleci.com/api/v2/project/gh/$OWNER/$REPOSITORY/pipeline?branch=$BRANCH" | jq -r ".items | map(select(.vcs.revision == \"${HEAD_COMMIT_HASH}\" )) | first | .id")
+          workflow_id=$(curl --silent "https://circleci.com/api/v2/pipeline/$pipeline_id/workflow" | jq -r ".items[0].id")
+          job_details=$(curl --silent "https://circleci.com/api/v2/workflow/$workflow_id/job" | jq -r '.items[] | select(.name == "job-publish-prerelease")')
+          build_num=$(echo "$job_details" | jq -r '.job_number')
+
+          echo 'CIRCLE_BUILD_NUM='"$build_num" >> "$GITHUB_OUTPUT"
+          job_id=$(echo "$job_details" | jq -r '.id')
+          echo 'CIRCLE_WORKFLOW_JOB_ID='"$job_id" >> "$GITHUB_OUTPUT"
+
+          echo "Getting artifacts from pipeline '${pipeline_id}', workflow '${workflow_id}', build number '${build_num}', job ID '${job_id}'"
+
+      - name: Get CircleCI job artifacts
```

Comment:
Most of the artifacts referenced in the `metamaskbot` comment are referenced by CircleCI artifact link. We didn't need to download those.

These three are pulled from the filesystem by the script though, so we needed to download these.